dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -300,28 +305,30 @@ private boolean problemFilter(Map.Entry<SymbolProblem, ClassFile> entry) {
 
       // The class is in class path but the symbol is not found
       return Optional.of(
-          new SymbolProblem(symbol, ErrorType.SYMBOL_NOT_FOUND, containingClassFile));
+          new LinkageProblem(
+              symbol, ErrorType.SYMBOL_NOT_FOUND, containingClassFile, sourceClassFile));
     } catch (ClassNotFoundException ex) {
       if (classDumper.catchesLinkageErrorOnClass(sourceClassName)) {
         return Optional.empty();
       }
       ClassSymbol classSymbol = new ClassSymbol(symbol.getClassBinaryName());
-      return Optional.of(new SymbolProblem(classSymbol, ErrorType.CLASS_NOT_FOUND, null));
+      return Optional.of(
+          new LinkageProblem(classSymbol, ErrorType.CLASS_NOT_FOUND, null, sourceClassFile));
```

Comment:
Here, if we define ClassNotFoundProblem then there's no need for the null third argument that doesn't apply

Reply:
That makes sense. Let me create these subclasses.